File: A's/2123A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int t;
    cin >> t;

    vector<string> winners;

    while (t--) {
        int n; cin >> n;

        if (n % 4 == 0) {
            winners.push_back("Bob");
        } else {
            winners.push_back("Alice");
        }
    }

    for (int i = 0; i < winners.size(); i++) {
        cout << winners.at(i) << endl;
    }
    
    return 0;
}

File: A's/py/2123A.py
def main() -> None:
    t: int = int(input())

    winners: list[str] = []

    while t:
        n: int = int(input())

        winners.append("Alice" if n % 4 != 0 else "Bob")

        t -= 1

    for w in winners:
        print(w)

if __name__ == "__main__":
    main()